bin/all_sky_search/pycbc_prepare_xml_for_gracedb

```diff
@@ -0,0 +1,216 @@
+parser.add_argument('--input-file', required=True, type=str,
+                    help='Input LIGOLW XML file of coincidences.')
+parser.add_argument('--event-id', type=int, required=True,
+                    help='ID for the event being prepared.')
+parser.add_argument('--output-file',
+                    help="Output filename for the locally stored XML. "
+                         "Default: current directory")
+parser.add_argument('--snr-timeseries-plot',
+                    help="Output filename for the plot of SNR timeseries. "
+                         "Must be a plot filetype as used by matplotlib.")
+parser.add_argument('--psd-plot',
+                    help="Output filename for the plot of the PSD. "
+                         "Must be a plot filetype as used by matplotlib.")
+parser.add_argument('--channel-name', action=MultiDetOptionAction,
+                    required=True,
+                    help="Channel name for adding to the uploaded single "
+                         "inspiral table.")
+parser.add_argument('--delta-f', type=float, default=0.25,
+                    help="Frequency spacing of the PSD to be uploaded, Hz. "
+                         "Default 0.25")
+
+args = parser.parse_args()
+
+if args.verbose:
+    args.verbose += 1
+else:
+    args.verbose = 1
+pycbc.init_logging(args.verbose)
+
+xmldoc = ligolw_utils.load_filename(args.input_file,
+                             contenthandler=LIGOLWContentHandler)
+
+class psd_segment(segment):
+    def __new__(cls, psd, *args):
+        return segment.__new__(cls, *args)
+    def __init__(self, psd, *args):
+        self.psd = psd
+
+logging.info("Reading PSDs")
+psds = {}
+for psd_file in args.psd_files:
+    (ifo, group), = h5py.File(psd_file, "r").items()
+    psd = [group["psds"][str(i)] for i in range(len(group["psds"].keys()))]
+    psds[ifo] = segmentlist(psd_segment(*segargs) for segargs in zip(
+        psd, group["start_time"], group["end_time"]))
+
+# Get information from the single_template snr timeseries files:
+logging.info("Reading SNR timeseries")
+snr_timeseries = {}
+for snr_filename in args.snr_timeseries:
+    with h5py.File(snr_filename, 'r') as snr_f:
+        ifo = snr_f.attrs['ifo']
+        time = snr_f.attrs['event_time']
+        snr_timeseries[(ifo, time)] = load_timeseries(snr_filename, 'snr')
+
+coinc_table = lsctables.CoincTable.get_table(xmldoc)
+coinc_inspiral_table = lsctables.CoincInspiralTable.get_table(xmldoc)
+coinc_event_map_table = lsctables.CoincMapTable.get_table(xmldoc)
+sngl_inspiral_table = lsctables.SnglInspiralTable.get_table(xmldoc)
+
+xmldoc.childNodes[-1].removeChild(sngl_inspiral_table)
+xmldoc.childNodes[-1].removeChild(coinc_event_map_table)
+xmldoc.childNodes[-1].removeChild(coinc_inspiral_table)
+xmldoc.childNodes[-1].removeChild(coinc_table)
+
+for event in coinc_table:
+    coinc_event_id = event.coinc_event_id
+    if args.event_id == coinc_event_id:
+        # This is the event we want
+        break
```

Comment:
I don't see how event could be None, as it would be changed in the loop?

Reply:
If the user provided `--event-id` is not actually contained in the XML, this will be detected by `event` being None. Without this change, the code would happily pick an effectively random event and proceed.

Reply:
I see that it would get `event` from the end of the loop in that case, which is bad, but the event would still not be None, it would be the last event in the table. I've added another variable to allow what you're thinking of though

Reply:
:picard: that was a major thinko, sorry. However I think Python has an even better solution: https://book.pythontips.com/en/latest/for_-_else.html

Reply:
Surprised I missed this thought - for/else is probably my favourite python feature (its a bit weird that I have one)